### ISSUE TITLE
Gradle runs with plain console output under MINTTY / Git Bash on Windows with no easy way to force rich output

### ISSUE THREAD

**Issue by zirzirikos** [gradle/gradle#2485]
Gradle, when run under the MINTTY console used by the MINGW/MSYS system (lately made very popular as it is the default command line environment for git on Windows), will use plain console output even though the terminal fully supports ncurses and rich console output. Since there is no system property that can be used in GRADLE_OPTS or ~/.gradle/gradle.properties to control this behaviour, the only option to force --console=rich is to use a shell alias, but that does not work with the gradle wrapper.

### Expected Behavior
Gradle using the default --console=auto setting should run with rich console output 
Alternatively there should be a way to force rich console output on a per-session basis that applies to wrapper scripts, preferably with a mechanism like GRADLE_OPTS

### Current Behavior
Gradle using the default --console=auto setting runs with plain console output 
Adding --console=rich to GRADLE_OPTS yields "Unrecognized option: --console=rich"

### Context
Cannot practically use rich console output and gradle wrapper on Windows / MsysGit.

### Steps to Reproduce (for bugs)
1. Install Git Bash available from https://git-for-windows.github.io/ 
2. Launch "git bash"
3. Run any gradle buiild

### Your Environment
Tested on Windows 7 x64 / msysgit 2.6.3 and 2.13.3

**Comment by eriwen** [comment]
I'm not sure if something in [native platform](https://github.com/adammurdoch/native-platform) thinks the MINTTY cannot handle rich output. It'd be nice to fix that, but saving that, I'd consider this issue a feature request for a persistent way to set rich console via Gradle properties.

**Comment by bigdaz** [comment]
Setting the `--console` value via `gradle.properties` is #2744

**Comment by binkley** [comment]
I find the same behavior for Cygwin, setting `--console=auto` does not produce rich output.  Setting `--console=rich` explicitly does.

**Comment by scharfstein-dev** [comment]
Anyone coming here trying to find out how to enable rich console output under Windows using Git Bash: This now works as per #2744. Simply set `org.gradle.console=rich` in your `gradle.properties` file. You will find it in your Gradle home, e.g. `%HOMEPATH%\.gradle\` (or simply create it).

**Comment by binkley** [comment]
@scharfstein-dev That is a practical solution, thank you.  Better would be for Gradle to recognize "Git for Windows", and "do the right thing".  Perhaps checking for some env var (?).  Some which jump out at me:
```
MINGW_CHOST=x86_64-w64-mingw32                                                   MINGW_PACKAGE_PREFIX=mingw-w64-x86_64
MINGW_PREFIX=/mingw64
MSYSTEM_CARCH=x86_64
MSYSTEM_CHOST=x86_64-w64-mingw32
MSYSTEM_PREFIX=/mingw64
MSYSTEM=MINGW64
```
The last one, `MSYSTEM=MINGW64`, seems like a reasonable choice.

**However** ... I'm surprised that the default `TERM=xterm` is not sufficient; that would certainly work without reliance on mingw-specific features.

**Comment by adammurdoch** [comment]
This is fixed in https://github.com/gradle/gradle/pull/8189. The fix will be in Gradle 5.2. Please create a new github issue if you run into any remaining issues with the console on Windows.
